check_run/overrides/sales_invoice.py

```diff
@@ -0,0 +1,61 @@
+import frappe
+from erpnext.accounts.doctype.accounting_dimension.accounting_dimension import (
+	get_accounting_dimensions,
+)
+from erpnext.accounts.doctype.sales_invoice.sales_invoice import SalesInvoice
+from erpnext.accounts.party import get_due_date
+from erpnext.accounts.utils import get_account_currency
+from frappe.utils.data import cint, flt
+
+
+class CheckRunSalesInvoice(SalesInvoice):
+	# TODO: update due date in taxes with validate hook
+	def validate(self):
+		super().validate()
+		for row in self.taxes:
+			if not row.party:
+				continue
+			due_date = get_due_date(self.posting_date, row.party_type, row.party, self.company)
+			row.due_date = due_date or self.posting_date
+			row.outstanding_amount = row.tax_amount
+
+	def on_submit(self):
+		for row in self.taxes:
+			if row.party and row.party_type:
+				row.outstanding_amount = row.tax_amount

```

Comment:
I don't understand why here the  `outstanding_amount` (and only this) is getting set again `on_submit`

Reply:
@fproldan @agritheory 
I have removed the code mentioned by @fproldan and did some testing. Without this code, the calculation of the outstanding amount is also working perfectly.

We can remove this on_submit function from the overridden file.